### PR TITLE
lab 3

### PR DESCRIPTION
**Pull request by igorv404**
Form verification:
Screenshot 2024-03-15 004343
User checking:
Screenshot 2024-03-15 004502
Get user data from device:
Screenshot 2024-03-15 004639
Edit user info:
Screenshot 2024-03-15 004737
Updated data:
Screenshot 2024-03-15 004901

### CODE REVIEW

File: lib/pages/edit.dart [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                                  snapshot.data!.countOfReservations;
+                              _secureStorage.saveUser(_newUser);
+                              Navigator.pushAndRemoveUntil(
+                                context,
+                                MaterialPageRoute(
+                                  builder: (context) => const AccountPage(),
+                                ),
+                                (route) => false,
+                              );
+                              // Navigator.pushAndRemoveUntil(context, '/account');
```
Comment (DenysDoskochynskyi) [inline lib/pages/edit.dart:115]: comments

File: lib/pages/login.dart [hunk 2]
```diff
@@ -29,6 +42,10 @@ class LoginPage extends StatelessWidget {
                         labelText: 'Enter your email',
                         border: OutlineInputBorder(),
                       ),
+                      validator: (email) => _validation.validateEmail(email!),
```
Comment (DenysDoskochynskyi) [inline lib/pages/login.dart:45]: reduce usage of `!`

File: lib/pages/registration.dart [hunk 3]
```diff
@@ -29,20 +38,36 @@ class RegistrationPage extends StatelessWidget {
                         labelText: 'Enter your name',
                         border: OutlineInputBorder(),
                       ),
+                      validator: (name) => name!.trim().length < 2
```
Comment (DenysDoskochynskyi) [inline lib/pages/registration.dart:41]: reduce usage of `!`

**Review by DenysDoskochynskyi — APPROVED** [review]
Nice work but please reduce usage of `!` because in future it can play with you bad joke)

File: lib/pages/edit.dart (hunk 1 above)
Comment (igorv404) [reply lib/pages/edit.dart]: Fixed

**Pull request merged**